clusterbuster

```diff
@@ -2427,6 +2440,65 @@ EOF
     done
 }
 
+function create_vm_deployment() {
+    local namespace=$1
+    local instance=$2
+    local -i replicas=$4
+    local -i replica=0
+    local depname="${namespace}-${workload}-${instance}"
+    while (( replica++ < replicas )) ; do
+        local name="${depname}-${replica}"
+        local vmname=$(mkpodname "$name")
+        create_object -n "$namespace" "$name" <<EOF
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  labels:
+    app: "$name"
+    kubevirt-vm: "$vmname"
+  name: "$vmname"
+$(indent 2 standard_deployment_metadata_yaml "$namespace" client)
+$(indent 2 standard_pod_metadata_yaml "$namespace" client)
+$(create_vm_spec -n "$vmname")
+EOF
+    done
+}
+
+function create_vm_spec() {
+    local vmname="fedora"
+    while getopts 'n:' opt; do
+        case "$opt" in
+            n) vmname=$OPTARG ;;
+        esac
+    done
+    [[ -n "$vmname" ]] || fatal "vmname is required"
+    cat <<EOF
+spec:
+  running: true
+  template:
+    metadata:
+      labels:
+        kubevirt-vm: "$vmname"
+    spec:
+      domain:
+        cpu:
+          cores: $cores
+          sockets: $sockets
+          threads: $threads
+        devices:
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio
+$(indent 8 container_resources_yaml)
+      terminationGracePeriodSeconds: $grace_period
+      volumes:
+        - name: containerdisk
+          containerDisk:
+            image: $image
```

Comment:
Perhaps the standard volumes code should also be used so that someone can mount other volumes within a VM.